custom_connectors/oauth2/ephesoft_connector.rb

```diff
@@ -0,0 +1,225 @@
+                      ).
+                      request_format_www_form_urlencoded
+        [
+          { # Tokens hash
+            access_token: response["access_token"],
+            refresh_token: response["refresh_token"]
+          },
+          { instance_id: nil } # Optional. Will be merged into connection hash
+        ]
+      end,
+
+      apply: lambda do |connection, access_token|
+        headers("Authorization": "Bearer #{access_token}")
+      end
+    }
+  },
+  test:
+    lambda do |connection|
+      get("https://api.us.ephesoft.io/v1/settings/integrations/configurations")
+    end,
+  actions: {
+    # Future action code here
+  },
+  triggers: {
+    new_message: {
+      title: 'When a Semantik invoice has completed review',
+      description: 'Triggers a flow using data from a completed Semantik invoice.',
+      webhook_subscribe: lambda do |webhook_url, connection, input, recipe_id|
+        post("https://api.us.ephesoft.io/v1/settings/integrations/configurations",
+          integrationName: "Workato Recipe #{recipe_id}",
+          integrationType: "webhook",
+          enabled: true,
+          settings: {
+            targetUrl: webhook_url,
+            encoding: "application/json",
+            payload: "{\n     \"AmountDue\": \"$AmountDue\",\n     \"CustomerId\": \"$CustomerId\",\n     \"DocumentId\": \"$DocumentId\",\n     \"DueDate\": \"$DueDate\",\n     \"InvoiceDate\": \"$InvoiceDate\",\n     \"InvoiceNumber\": \"$InvoiceNumber\",\n     \"OrderDate\": \"$OrderDate\",\n     \"PdfUrl\": \"$PdfUrl\",\n     \"PONumber\": \"$PONumber\",\n     \"ShipDate\": \"$ShipDate\",\n     \"ShipFreight\": \"$ShipFreight\",\n     \"SubTotal\": \"$SubTotal\",\n     \"TableUrl\": \"$TableUrl\",\n     \"TaxAmount\": \"$TaxAmount\",\n     \"TaxId\": \"$TaxId\",\n     \"TaxRate\": \"$TaxRate\",\n     \"TenantId\": \"$TenantId\",\n     \"Terms\": \"$Terms\",\n     \"TotalAmount\": \"$TotalAmount\",\n     \"Vendor\": {\n       \"VendorID\": \"$Vendor:VendorID\",\n       \"VendorName\": \"$Vendor:Name\",\n       \"VendorAddress\": {\n         \"VendorStreetAddress\": \"$Vendor:StreetAddress\",\n         \"VendorPOBox\": \"$Vendor:POBox\",\n         \"VendorLocality\": \"$Vendor:Locality\",\n         \"VendorRegion\": \"$Vendor:Region\",\n         \"VendorPostalCode\": \"$Vendor:PostalCode\",\n         \"VendorCountry\": \"$Vendor:Country\"\n       },\n       \"VendorTelephone\": \"$Vendor:Telephone\",\n       \"VendorCustomerID\": \"$Vendor:CustomerID\",\n       \"VendorStatus\": \"$Vendor:Status\",\n       \"VendorSWIFT\": \"$Vendor:SWIFT\",\n       \"VendorIBAN\": \"$Vendor:IBAN\",\n       \"VendorCustom1\": \"$Vendor:Custom1\",\n       \"VendorCustom2\": \"$Vendor:Custom2\",\n       \"VendorCustom3\": \"$Vendor:Custom3\",\n       \"VendorCustom4\": \"$Vendor:Custom4\",\n       \"VendorCustom5\": \"$Vendor:Custom5\",\n       \"VendorMatched\": \"$Vendor:Matched\"\n     }\n}"
```

Comment:
One way to make this easier to maintain. I believe you can declare the payload in its natural json format and then stringify it.

```
payload = {
"AmountDue": "$AmountDue"
# more keys here
}

post("https://api.us.ephesoft.io/v1/settings/integrations/configurations",
          integrationName: "Workato Recipe #{recipe_id}",
          integrationType: "webhook",
          enabled: true,
          settings: {
            targetUrl: webhook_url,
            encoding: "application/json",
            payload: payload.to_json.to_s
           }
        )
```

Reply:
Might be easier to add more keys if you need to in the future.

Reply:
Yeah, this is much better.  Updated in latest commit.